src/realm/sync/transform.cpp

```diff
@@ -1597,6 +1635,20 @@ DEFINE_MERGE(Instruction::Update, Instruction::Update)
             REALM_MERGE_ASSERT(left.prior_size == right.prior_size);
         }
 
+        if (left.value.type != right.value.type) {
+            // Embedded object / dictionary creation should always lose to an
+            // Update(NULL), because these structures are nested, and we need to
+            // discard any update inside the structure.
+            if (left.value.type == Type::Dictionary || left.value.type == Type::ObjectValue) {
```

Comment:
Also, we want to change these to include Type::Set as well, right?

Reply:
Sets cannot contain nested structures (because that risks breaking the uniqueness invariant).